File: 백준/Silver/4673. 셀프 넘버/셀프 넘버.py
import sys

self_num = set(range(1, 10001))

for i in range(1, 10001):
    num = i
    tmp = num

    while tmp:
        num += tmp % 10
        tmp //= 10
        
    if num in self_num:
        self_num.remove(num)

for i in sorted(self_num):
    sys.stdout.write(f"{i}\n")

File: 백준/Silver/4673. 셀프 넘버/셀프 넘버.cc
#include <iostream>
#include <vector>
#include <algorithm>
 using namespace std;

int main() {
    ios::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);
    
    vector<bool> self_num(10001, true);

    for (int i = 1; i <= 10000; i++) {
        int num = i;
        int tmp = num;

        while (tmp) {
            num += tmp % 10;
            tmp /= 10;
        }

        if (num <= 10000)
            self_num[num] = false;
    }

    for (int i = 1; i <= 10000; i++) {
        if (self_num[i])
            cout << i << "\n";
    }
    
    return 0;
}